var/spack/repos/builtin/packages/mesa/package.py

```diff
@@ -30,16 +30,29 @@ class Mesa(AutotoolsPackage):
     specification - a system for rendering interactive 3D graphics."""
 
     homepage = "http://www.mesa3d.org"
-    url      = "http://ftp.iij.ad.jp/pub/X11/x.org/pub/mesa/12.0.3/mesa-12.0.3.tar.gz"
+    base_url = "ftp://ftp.freedesktop.org/pub/mesa/"
+    url      = base_url+"12.0.3/mesa-12.0.3.tar.gz"
 
     version('12.0.3', '60c5f9897ddc38b46f8144c7366e84ad')
-
+    version('13.0.2', 'cb828f9f68e0cd6c7ef30b9c67aa7cf2')
```

Comment:
please sort versions in descending order, i.e. `13.0.2` first, then `13.0.1`, etc...